src/hdmf_zarr/backend.py

```diff
@@ -783,29 +817,26 @@ def write_dataset(self, **kwargs):  # noqa: C901
                 for j, item in enumerate(data):
                     new_item = list(item)
                     for i in refs:
-                        new_item[i] = self.__get_ref(item[i])
+                        new_item[i] = self.__get_ref(item[i], export_source=export_source)
                     dset[j] = new_item
             else:
                 # write a compound datatype
                 dset = self.__list_fill__(parent, name, data, options)
         # Write a dataset of references
         elif self.__is_ref(options['dtype']):
-            if isinstance(data, RegionBuilder):
-                shape = (1,)
-                type_str = 'region'
-                refs = self.__get_ref(data.builder, data.region)
-            elif isinstance(data, ReferenceBuilder):
+            if isinstance(data, ReferenceBuilder):
```

Comment:
```suggestion
            # TODO Region references are not yet support, but here how the code should look
            #  if isinstance(data, RegionBuilder):
            #      shape = (1,)
            #      type_str = 'region'
            #      refs = self.__get_ref(data.builder, data.region)
            if isinstance(data, ReferenceBuilder):
```

I know I originally suggested removing this comment. But I would keep it, because it will make implementing region references easier in the future.